### PR TITLE
fix: axonaut verfication endpoint fix

### LINKED ISSUE #3449: Issue with incorrect verification endpoint for Axonaut
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### Expected Behavior
The API endpoint used for verification should be `/api/v2/me` instead of current endpoint.
Reference : https://axonaut.com/api/v2/doc

### Actual Behavior
The current endpoint for API key verification is `https://axonaut.com/api/v2/companies?type=all&sort=id` which is not recommended, should be `/api/v2/me`. 

## Environment
 * OS: [e.g. iOS]
 * Version [e.g. 22]

### References

* #0000

### PR DESCRIPTION
**Pull request by sahil9001**
### Description:
Fixes #3449 .
Screenshot 2024-10-17 at 1 18 05 PM

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

**Comment by sahil9001** [comment]
@zricethezav @rgmz

File: pkg/detectors/axonaut/axonaut.go [hunk 1]
```diff
@@ -47,7 +48,8 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", "https://axonaut.com/api/v2/companies?type=all&sort=id", nil)
+			// Reference : https://axonaut.com/api/v2/doc
+			req, err := http.NewRequestWithContext(ctx, "GET", "https://axonaut.com/api/v2/me", nil)
```
Comment (kashifkhan0771) [inline pkg/detectors/axonaut/axonaut.go:52]: The API returns user email which we can add in extra data but the API also return some other sensitive information about user. I'll let @mcastorina @abmussani comment here.

**Comment by sahil9001** [comment]
@mcastorina @abmussani Can you please review this one?

**Comment by zricethezav** [comment]
This API change isn't fixing anything since `https://axonaut.com/api/v2/companies?type=all&sort=id` is a valid API endpoint to test verification. Closing

**Pull request closed without merging**